dockerfiles/executor_image/Dockerfile

```diff
@@ -15,9 +15,32 @@ RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add - && \
 # Also install iproute2 ("ip" command) to configure networking on host.
 RUN apt-get update && apt-get install -y umoci iproute2
 
-RUN echo 'deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/Debian_11/ /' >> /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list && \
-    curl -fsSL https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/Debian_11/Release.key | gpg --dearmor >> /etc/apt/trusted.gpg.d/devel_kubic_libcontainers_stable.gpg && \
-    apt-get update && apt-get -y upgrade && apt-get install -y podman skopeo
+# Install Podman 4
+# Install Podman dependencies
+RUN apt-get update && apt-get install -y \
+  btrfs-progs git go-md2man iptables libassuan-dev libbtrfs-dev libc6-dev libdevmapper-dev libglib2.0-dev libgpgme-dev libgpg-error-dev libprotobuf-dev libprotobuf-c-dev libseccomp-dev libselinux1-dev libsystemd-dev pkg-config uidmap
+RUN apt-get update && apt-get install -y make build-essential
+# Install golang-go. Requires 1.16+
+RUN curl -fsSL "https://go.dev/dl/go1.17.8.linux-amd64.tar.gz" | tar xz -C /usr/local/
+ENV PATH="/usr/local/go/bin:${PATH}"
+RUN curl -OL "https://github.com/opencontainers/runc/releases/download/v1.1.0/runc.amd64" && \ 
+    chmod a+x runc.amd64 && mv runc.amd64 /usr/bin/runc
+# Install podman
+RUN curl -fsSL "https://github.com/containers/podman/archive/refs/tags/v4.0.2.tar.gz" | tar xz 
+RUN cd podman-4.0.2 && make BUILDTAGS="selinux seccomp"
+RUN cd podman-4.0.2 && make install PREFIX=/usr
+# Install conmon
+RUN curl -fsSL https://github.com/containers/conmon/archive/refs/tags/v2.1.0.tar.gz | tar xz
+RUN cd conmon-2.1.0 && make
+RUN cd conmon-2.1.0 && make podman
+# Install cni-plugins
+RUN mkdir -p /opt/cni/bin
+RUN curl -fsSL https://github.com/containernetworking/plugins/releases/download/v1.1.1/cni-plugins-linux-amd64-v1.1.1.tgz | tar xz -C /opt/cni/bin/
+
+# Add configuration
+RUN mkdir -p /etc/containers
+RUN curl -L -o /etc/containers/registries.conf https://src.fedoraproject.org/rpms/containers-common/raw/main/f/registries.conf
+RUN curl -L -o /etc/containers/policy.json https://src.fedoraproject.org/rpms/containers-common/raw/main/f/default-policy.json
```

Comment:
I think each time you add a "RUN" command docker is going to make a new layer. Is it possible to add the --squash flag to the build step (in the BUILD file?)